File: bindings/lua/Polycode/Sound.lua
class "Sound"


function Sound:Sound(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.Sound(unpack(arg))
	end
end

function Sound:getSampleAsNumber(offset, channel, position, orientation)
	local retVal = Polycode.Sound_getSampleAsNumber(self.__ptr, offset, channel, position.__ptr, orientation.__ptr)
	return retVal
end

function Sound:loadFile(fileName)
	local retVal = Polycode.Sound_loadFile(self.__ptr, fileName)
end

function Sound:Play(loop, restartSound)
	local retVal = Polycode.Sound_Play(self.__ptr, loop, restartSound)
end

function Sound:Stop()
	local retVal =  Polycode.Sound_Stop(self.__ptr)
end

function Sound:setVolume(newVolume)
	local retVal = Polycode.Sound_setVolume(self.__ptr, newVolume)
end

function Sound:getVolume()
	local retVal =  Polycode.Sound_getVolume(self.__ptr)
	return retVal
end

function Sound:setPitch(newPitch)
	local retVal = Polycode.Sound_setPitch(self.__ptr, newPitch)
end

function Sound:getPitch()
	local retVal =  Polycode.Sound_getPitch(self.__ptr)
	return retVal
end

function Sound:isPlaying()
	local retVal =  Polycode.Sound_isPlaying(self.__ptr)
	return retVal
end

function Sound:isLooped()
	local retVal =  Polycode.Sound_isLooped(self.__ptr)
	return retVal
end

function Sound:setIsPositional(isPositional)
	local retVal = Polycode.Sound_setIsPositional(self.__ptr, isPositional)
end

function Sound:setSoundPosition(position)
	local retVal = Polycode.Sound_setSoundPosition(self.__ptr, position.__ptr)
end

function Sound:setSoundVelocity(velocity)
	local retVal = Polycode.Sound_setSoundVelocity(self.__ptr, velocity.__ptr)
end

function Sound:setSoundDirection(direction)
	local retVal = Polycode.Sound_setSoundDirection(self.__ptr, direction.__ptr)
end

function Sound:setOffset(offset)
	local retVal = Polycode.Sound_setOffset(self.__ptr, offset)
end

function Sound:getFileName()
	local retVal =  Polycode.Sound_getFileName(self.__ptr)
	return retVal
end

function Sound:getPlaybackDuration()
	local retVal =  Polycode.Sound_getPlaybackDuration(self.__ptr)
	return retVal
end

function Sound:getPlaybackTime()
	local retVal =  Polycode.Sound_getPlaybackTime(self.__ptr)
	return retVal
end

function Sound:seekTo(time)
	local retVal = Polycode.Sound_seekTo(self.__ptr, time)
end

function Sound:getOffset()
	local retVal =  Polycode.Sound_getOffset(self.__ptr)
	return retVal
end

function Sound:getSampleLength()
	local retVal =  Polycode.Sound_getSampleLength(self.__ptr)
	return retVal
end

function Sound:getFrequency()
	local retVal =  Polycode.Sound_getFrequency(self.__ptr)
	return retVal
end

function Sound:setPositionalProperties(referenceDistance, maxDistance)
	local retVal = Polycode.Sound_setPositionalProperties(self.__ptr, referenceDistance, maxDistance)
end

function Sound:setReferenceDistance(referenceDistance)
	local retVal = Polycode.Sound_setReferenceDistance(self.__ptr, referenceDistance)
end

function Sound:setMaxDistance(maxDistance)
	local retVal = Polycode.Sound_setMaxDistance(self.__ptr, maxDistance)
end

function Sound:getReferenceDistance()
	local retVal =  Polycode.Sound_getReferenceDistance(self.__ptr)
	return retVal
end

function Sound:getMaxDistance()
	local retVal =  Polycode.Sound_getMaxDistance(self.__ptr)
	return retVal
end

function Sound:loadWAV(fileName)
	local retVal = Polycode.Sound_loadWAV(self.__ptr, fileName)
	return retVal
end

function Sound:loadOGG(fileName)
	local retVal = Polycode.Sound_loadOGG(self.__ptr, fileName)
	return retVal
end

function Sound:soundCheck(result, err)
	local retVal = Polycode.Sound_soundCheck(self.__ptr, result, err)
end

function Sound:updateStream(streamCount)
	local retVal = Polycode.Sound_updateStream(self.__ptr, streamCount)
end

function Sound:__delete()
	if self then Polycode.delete_Sound(self.__ptr) end
end

File: bindings/javascript/Polycode/Sound.js
function Sound(fileName) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.Sound(fileName)
	}
}

Duktape.fin(Sound.prototype, function (x) {
	if (x === Sound.prototype) {
		return;
	}
	Polycode.Sound__delete(x.__ptr)
})

Sound.prototype.getSampleAsNumber = function(offset,channel,position,orientation) {
	return Polycode.Sound_getSampleAsNumber(this.__ptr, offset, channel, position, orientation)
}

Sound.prototype.loadFile = function(fileName) {
	Polycode.Sound_loadFile(this.__ptr, fileName)
}

Sound.prototype.Play = function(loop,restartSound) {
	Polycode.Sound_Play(this.__ptr, loop, restartSound)
}

Sound.prototype.Stop = function() {
	Polycode.Sound_Stop(this.__ptr)
}

Sound.prototype.setVolume = function(newVolume) {
	Polycode.Sound_setVolume(this.__ptr, newVolume)
}

Sound.prototype.getVolume = function() {
	return Polycode.Sound_getVolume(this.__ptr)
}

Sound.prototype.setPitch = function(newPitch) {
	Polycode.Sound_setPitch(this.__ptr, newPitch)
}

Sound.prototype.getPitch = function() {
	return Polycode.Sound_getPitch(this.__ptr)
}

Sound.prototype.isPlaying = function() {
	return Polycode.Sound_isPlaying(this.__ptr)
}

Sound.prototype.isLooped = function() {
	return Polycode.Sound_isLooped(this.__ptr)
}

Sound.prototype.setIsPositional = function(isPositional) {
	Polycode.Sound_setIsPositional(this.__ptr, isPositional)
}

Sound.prototype.setSoundPosition = function(position) {
	Polycode.Sound_setSoundPosition(this.__ptr, position)
}

Sound.prototype.setSoundVelocity = function(velocity) {
	Polycode.Sound_setSoundVelocity(this.__ptr, velocity)
}

Sound.prototype.setSoundDirection = function(direction) {
	Polycode.Sound_setSoundDirection(this.__ptr, direction)
}

Sound.prototype.setOffset = function(offset) {
	Polycode.Sound_setOffset(this.__ptr, offset)
}

Sound.prototype.getFileName = function() {
	return Polycode.Sound_getFileName(this.__ptr)
}

Sound.prototype.getPlaybackDuration = function() {
	return Polycode.Sound_getPlaybackDuration(this.__ptr)
}

Sound.prototype.getPlaybackTime = function() {
	return Polycode.Sound_getPlaybackTime(this.__ptr)
}

Sound.prototype.seekTo = function(time) {
	Polycode.Sound_seekTo(this.__ptr, time)
}

Sound.prototype.getOffset = function() {
	return Polycode.Sound_getOffset(this.__ptr)
}

Sound.prototype.getSampleLength = function() {
	return Polycode.Sound_getSampleLength(this.__ptr)
}

Sound.prototype.getFrequency = function() {
	return Polycode.Sound_getFrequency(this.__ptr)
}

Sound.prototype.setPositionalProperties = function(referenceDistance,maxDistance) {
	Polycode.Sound_setPositionalProperties(this.__ptr, referenceDistance, maxDistance)
}

Sound.prototype.setReferenceDistance = function(referenceDistance) {
	Polycode.Sound_setReferenceDistance(this.__ptr, referenceDistance)
}

Sound.prototype.setMaxDistance = function(maxDistance) {
	Polycode.Sound_setMaxDistance(this.__ptr, maxDistance)
}

Sound.prototype.getReferenceDistance = function() {
	return Polycode.Sound_getReferenceDistance(this.__ptr)
}

Sound.prototype.getMaxDistance = function() {
	return Polycode.Sound_getMaxDistance(this.__ptr)
}

Sound.prototype.loadBytes = function(data,size,channels,freq,format) {
	return Polycode.Sound_loadBytes(this.__ptr, data.__ptr, size, channels, freq, format)
}

Sound.prototype.loadWAV = function(fileName) {
	return Polycode.Sound_loadWAV(this.__ptr, fileName)
}

Sound.prototype.loadOGG = function(fileName) {
	return Polycode.Sound_loadOGG(this.__ptr, fileName)
}

Sound.prototype.soundCheck = function(result,err) {
	Polycode.Sound_soundCheck(this.__ptr, result, err)
}

Sound.prototype.updateStream = function(streamCount) {
	Polycode.Sound_updateStream(this.__ptr, streamCount)
}
